### PR TITLE
[ARCTIC-712][AMS] Support for regular expressions to match database name for Hive Metastore

### LINKED ISSUE #712: [Improvement]: Support for regular expressions to match database name for Hive Metastore
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/arctic/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

The existing HMS has a large number of hive tables, and all the databases and tables have to be loaded each time. 
Support for regular matching of database

### How should we improve?

Add an input when creating a catalog to support regular expressions to match database names

### Are you willing to submit PR?

- [x] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhongqishang**
## Why are the changes needed?

Fix #712 

## Brief change log

  - *Add `include.databases` in `CatalogMetaProperties`.*
  - *Add regular expressions to match database name in `com.netease.arctic.catalog.BaseIcebergCatalog#listDatabases`*

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (yes)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: ams/ams-api/src/main/java/com/netease/arctic/ams/api/properties/CatalogMetaProperties.java [hunk 1]
```diff
@@ -38,6 +38,7 @@ public class CatalogMetaProperties {
     public static final String AUTH_CONFIGS_VALUE_TYPE_KERBEROS = "kerberos";
 
     public static final String KEY_WAREHOUSE_DIR = "warehouse.dir";
+    public static final String KEY_INCLUDE_DATABASES = "include.databases";
```
Comment (zhoujinsong) [inline ams/ams-api/src/main/java/com/netease/arctic/ams/api/properties/CatalogMetaProperties.java:41]: How about changing the property name to `database.filter-regular-expression`?

File: core/src/main/java/com/netease/arctic/catalog/BaseIcebergCatalog.java [hunk 2]
```diff
@@ -70,14 +72,24 @@ public void initialize(
...
+        tableMetaStore.doAs(() ->
+            ((SupportsNamespaces) icebergCatalog).listNamespaces(Namespace.empty())
+                .stream()
+                .map(namespace -> namespace.level(0))
+                .distinct()
+                .collect(Collectors.toList())
+        );
+    String filter = meta.getCatalogProperties().get(CatalogMetaProperties.KEY_INCLUDE_DATABASES);
+    return databases.stream()
+        .filter(database -> filter == null || Pattern.matches(filter, database))
```
Comment (zhoujinsong) [inline core/src/main/java/com/netease/arctic/catalog/BaseIcebergCatalog.java:91]: Maybe we can add a class variable `databaseFilterPattern` and initialize it in method `initialize `, so we will not need to read these properties and construct Pattern every time?

**Review by zhoujinsong — COMMENTED** [review]
@zhongqishang Thanks a lot for your contribution, I left some comments, please take another look.

File: core/src/main/java/com/netease/arctic/catalog/BaseIcebergCatalog.java (hunk 2 above)
Comment (zhongqishang) [reply core/src/main/java/com/netease/arctic/catalog/BaseIcebergCatalog.java]: done.

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Pull request merged**